src/state/persist_deleted.go

```diff
@@ -0,0 +1,29 @@
+package state
+
+import (
+	"keboola-as-code/src/local"
+	"keboola-as-code/src/manifest"
+	"keboola-as-code/src/utils"
+)
+
+// PersistDeleted objects from the filesystem
+func (s *State) PersistDeleted() (deleted []manifest.Record, err error) {
+	errors := utils.NewMultiError()
+
+	// Search for deleted objects
+	records := s.manifest.GetRecords()
+	for _, key := range append([]string(nil), records.Keys()...) {
```

Comment:
tu je zaludnost, .... `records.Keys` sa musia pred iterovanim skopirovat, pretoze vo `for` slucke sa zaznamy mazu, a dana `slice` sa modifikuje.